Zend/zend_fibers.c

```diff
@@ -443,7 +443,7 @@ static ZEND_STACK_ALIGNED void zend_fiber_execute(zend_fiber_transfer *transfer)
 	zend_fiber *fiber = EG(active_fiber);
 
 	/* Determine the current error_reporting ini setting. */
-	zend_long error_reporting = INI_INT("error_reporting");
+	int error_reporting = INI_INT("error_reporting");
```

Comment:
`zend_ini_long()` returns `zend_long`, so why should we use a narrower type here?

Reply:
This value is later assigned to `EG(error_reporting), an `int`, so I suggested this change thinking perhaps there was something unusual happening where it wasn't being cast properly on that assignment because it was stored in a long.

Thinking about it further, I'm not sure this is origin of the problem anyway. The problem appears to be with saving and restoring `EG(error_reporting)`, which is done within `zend_fiber_capture_vm_state` and `zend_fiber_restore_vm_state`. Given the types there, I'm not seeing why this is failing.

Reply:
wondering as `git grep INI_INT` returns mostly `int` so it needs separate PR to normalize

btw gonna try 5459ed4c2f as it the fixed looks related

Reply:
but it does not help too

Reply:
As tests fixed, the remaining question is clean-up still needed?

Reply:
Commit 492af9f88e adds new function for parsing ints

Reply:
Somehow it still needs to fix type or use cast

Reply:
At least counters could affect all little-endian arches

Reply:
@cmb69 any suggestion to fix it?